openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ParametersGenerator.java

```diff
@@ -536,7 +541,7 @@ private ArrayTypeDescriptorNode getArrayTypeDescriptorNode(Schema<?> items) thro
                 createToken(SyntaxKind.CLOSE_BRACKET_TOKEN));
         NodeList<ArrayDimensionNode> nodeList = createNodeList(dimensionNode);
 
-        if (items.getNullable() != null && items.getNullable()) {
+        if (items.getNullable() != null && items.getNullable() && (items.getEnum() == null)) {
```

Comment:
```suggestion
        if (items.getNullable() != null && items.getNullable() && items.getEnum() == null) {
```

Reply:
Updated in https://github.com/ballerina-platform/openapi-tools/compare/e6163967946301919f5de65a2299cc1d3d2f8447..869fb96c762f2993035139df2763eed6713faa9c